tiatoolbox/tools/pyramid.py

```diff
@@ -430,3 +433,251 @@ def tile_path(self, level: int, x: int, y: int) -> Path:
+            >>> tile_generator = AnnotationTileGenerator(
+            ...   info=wsi.info,
+            ...   store=SQ,
+            ... )
+            >>> tile_0_0_0 = tile_generator.get_tile(level=0, x=0, y=0)
+
+        """
+        if pad_mode is not None or interpolation is not None:
+            warnings.warn(
+                "interpolation, pad_mode are unused by AnnotationTileGenerator"
+            )
+        if level < 0:
+            raise IndexError
+        if level > self.level_count:
+            raise IndexError("Invalid level.")
+
+        scale = self.level_downsample(level)
+        baseline_x = (x * self.tile_size * scale) - (self.overlap * scale)
+        baseline_y = (y * self.tile_size * scale) - (self.overlap * scale)
+        output_size = [self.output_tile_size] * 2
+        coord = [baseline_x, baseline_y]
+        if level < self.sub_tile_level_count:
+            output_size = self.output_tile_size // 2 ** (
+                self.sub_tile_level_count - level
+            )
+            output_size = np.repeat(output_size, 2).astype(int)
+            thumb = self.get_thumb_tile()
+            thumb.thumbnail(output_size)
+            return thumb
+        slide_dimensions = np.array(self.info.slide_dimensions)
+        if all(slide_dimensions < [baseline_x, baseline_y]):
+            raise IndexError
+
+        rgb = np.zeros((*output_size, 4), dtype=np.uint8)
+        bounds = locsize2bounds(coord, [self.tile_size * scale] * 2)
+        bound_geom = Polygon.from_bounds(*bounds)
+        rgb = self.render_annotations(rgb, bound_geom, scale, coord)
+
+        return Image.fromarray(rgb)
+
+    def render_by_type(
+        self,
+        rgb: np.ndarray,
+        ann: Annotation,
+        top_left: Tuple[float, float],
+        scale: int,
+        poly_as_box: bool = False,
+    ):
+        """Render annotation appropriately to its geometry type.
+
+        Args:
+            rgb (np.ndarray):
+                The image to render the annotation on.
+            ann (Annotation):
+                The annotation to render.
+            top_left (Tuple[int, int]):
+                The top left coordinate of the tile.
+            scale (int):
+                The scale at which we are rendering the tile.
+            poly_as_box (bool):
+                Whether to render polygons as boxes.
+
+        """
+        r = self.renderer
+        geom_type = ann.geometry.geom_type
+        if geom_type == "Point":
+            r.render_pt(rgb, ann, top_left, scale)
+        elif geom_type == "Polygon":
+            if poly_as_box:
+                r.render_rect(rgb, ann, top_left, scale)
+            else:
+                r.render_poly(rgb, ann, top_left, scale)
+        elif "Line" in geom_type:
+            r.render_line(rgb, ann, top_left, scale)
+        else:
+            warnings.warn("Unknown geometry")
+
+    def render_annotations(
+        self,
+        rgb: np.ndarray,
```

Comment:
1. Does this have to be RGB or will it also work with greyscale or RGBA?
2. I could call this image instead of RGB, as RGB sounds like you are supposed to give a colour to be e.g. `(255, 0, 255)`.

Reply:
It will work with rgb or rgba. It wouldnt work with grayscale.

I called it rgb for consistency with pyramid/zoomify generator.
Can change it to image if desired but would be best to change it in both places in that case.